File: src/main/Java/boj_step7/boj10798.java
import java.io.*;

public class boj10798 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        char[][] list = new char[5][15];
        for (int i = 0; i < list.length; i++) {
            char[] c = reader.readLine().toCharArray();
            for (int j = 0; j < c.length; j++) list[i][j] = c[j];
        }
        for (int j = 0; j < 15; j++)
            for (int i = 0; i < list.length; i++)
                if (list[i][j] != 0)
                    writer.append(list[i][j]);
        writer.close();
    }
}

File: src/main/Kotlin/others/boj10798.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import kotlin.math.max

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val arr = Array<String>(5) { "" }
    var maxLen = 0
    for (index in arr.indices) {
        arr[index] = r.readLine()!!
        maxLen = max(maxLen, arr[index].length)
    }
    for (i in 0 until maxLen)
        for (j in arr.indices) if (arr[j].getOrNull(i) != null) w.write(arr[j].get(i).toString()) else continue
    w.close()
}